controllers/component_controller.go

```diff
@@ -301,6 +301,13 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		var devfileBytes []byte
 
 		if source.GitSource != nil && source.GitSource.URL != "" {
+			if err := util.ValidateGithubURL(source.GitSource.URL); err != nil {
+				// User error - the git url provided is not from github
+				log.Error(err, "unable to validate github url")
+				metrics.IncrementComponentCreationSucceeded("", "")
+				_ = r.SetCreateConditionAndUpdateCR(ctx, req, &component, err)
+				return ctrl.Result{}, nil
+			}
```

Comment:
I think this will still be executed via HAC UI? Can you confirm by just building the HAS image and creating a Component via CLI?

We should not block gitlab errors out like returning over here, ref https://redhat-internal.slack.com/archives/C02HZRGKDEY/p1711484683757149?thread_ts=1711484510.936249&cid=C02HZRGKDEY

I think we should have a flag or something which indicates its a Gitlab repo and anytime there is an error with accessing the Gitlab repo because we are using Github API, then we can ignore that in our metrics count and return as a normal user error..

@johnmcollier thoughts?

Reply:
Yeah, I'd avoid returning here, and just let the controller handle the URL as it was before.

Reply:
@johnmcollier & @maysunfaisal I've tried to remove the early return and add the `gitSourceFromGitlab`.

Tested locally to create a `component` with gitlab URL and had got `"Component create failed: unable to get default branch of Github Repo"` error which is the "normal" error we got in the past. The difference here is that if the `gitSourceFromGitlab` is `true` we increase the successful component creation metric instead of the component creation failed one.

I've updated the test cases to check the metrics count is ok.